notebooks/utils/phenopacket.py

```diff
@@ -0,0 +1,199 @@
+
+    A valid phenopacket must be a dict containing:
+      - "phenotypicFeatures": a list of feature objects, each with a "type" dict:
+            {
+              "id": "HP:0000001",
+              "label": "Phenotype label"
+            }
+    """
+
+    def __init__(self, phenopacket_json: Dict[str, Any]) -> None:
+        """
+        Initialize from a JSON-decoded dict.
+
+        Parameters
+        ----------
+        phenopacket_json : dict
+            Parsed JSON object for a phenopacket.
+
+        Raises
+        ------
+        InvalidPhenopacketError
+            If the input is not a dict or phenotypicFeatures is malformed.
+        """
+        if not isinstance(phenopacket_json, dict):
+            raise InvalidPhenopacketError("Phenopacket JSON must be a dict.")
+        phenotypicFeatures = phenopacket_json.get("phenotypicFeatures", [])
+        if not isinstance(phenotypicFeatures, list):
+            raise InvalidPhenopacketError("`phenotypicFeatures` must be a list.")
+        self._json: Dict[str, Any] = phenopacket_json
+        self._phenotypicFeatures: List[Dict[str, Any]] = phenotypicFeatures
+
+    @classmethod
+    def load_from_file(cls, path: str) -> "Phenopacket":
+        """
+        Load a phenopacket JSON file from disk.
+
+        Parameters
+        ----------
+        path : str
+            Path to a .json file containing a GA4GH phenopacket.
+
+        Returns
+        -------
+        Phenopacket
+            Instance wrapping the loaded JSON.
+
+        Raises
+        ------
+        FileNotFoundError
+            If the given path does not exist.
+        json.JSONDecodeError
+            If the file cannot be parsed as JSON.
+        InvalidPhenopacketError
+            If the loaded JSON has the wrong structure.
+        """
+        if not os.path.isfile(path):
+            raise FileNotFoundError(f"No such file: {path}")
+        with open(path, "r", encoding="utf-8") as f:
+            data = json.load(f)
+        return cls(data)
+
+    def contains_phenotype(self, hpo_label: str) -> bool:
+        """
+        Check for presence of a phenotype by its human-readable label.
+
+        Parameters
+        ----------
+        hpo_label : str
+            The `type.label` of the phenotype to search for.
+
+        Returns
+        -------
+        bool
+            True if any feature has a matching label, False otherwise.
+        """
+        for feat in self._phenotypicFeatures:
+            if not isinstance(feat, dict):
+                logger.warning("Skipping malformed phenotypicFeature: not a dict")
+                continue
+            term = feat.get("type")
+            if not isinstance(term, dict):
+                logger.warning(
+                    "Skipping malformed phenotypicFeature: no or non-dict `type` field"
+                )
+                continue
+            if term.get("label") == hpo_label:
+                return True
+        return False
```

Comment:
I think the whole function can be:

```suggestion
hpo_label in self.list_phenotypes()
```